### PR TITLE
LPS-200125 Return the base64 encoded data of an Objects attachments field if it is required as nested field

### PR DESCRIPTION
**Pull request by ccorreagg**
Related ticket: https://liferay.atlassian.net/browse/LPS-200125

---

The purpose of this PR is to return the base64 encoded data of an attachment field of a given Object Entry in the `GET` endpoints. For that, the user has to require the field by using the query parameter `nestedFields`.

Given an attachmenf field called `myAttachmentField`, the format of the field in the Object Entry REST API is the following one:

```
"myAttachmentField": {
  "id": 1234,
  "link": {
    "href": "/documents/20113/0/ticket_16.10.2023.jpg/....",
    "label": "image.jpg"
  },
  "name": "image.jpg",
}
```
For the new base64 encoded content, we have created the field `fileBase64` inside the `FileEntry` schema:

```
"myAttachmentField": {
  "fileBase64": "<base64 content>",
  "id": 1234,
  "link": {
    "href": "/documents/20113/0/ticket_16.10.2023.jpg/....",
    "label": "image.jpg"
  },
  "name": "image.jpg"
}
```

In order to get the new `fileBase64` subfield, the format of the nested field is:

`nestedFields=<name_of_the_field>.fileBase64`, in this case `nestedFields=myAttachmentField.fileBase64`

### CODE REVIEW

**Comment by ccorreagg** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -28,6 +28,12 @@ components:
                     type: object
         FileEntry:
             properties:
+                fileBase64:
+                    # @review
+                    description:
+                        optional field with the content of the document in Base64, can be embedded with nestedFields
```
Comment (4lejandrito) [inline modules/apps/object/object-rest-impl/rest-openapi.yaml:34]: @ccorreagg how will users know what to set as nested field? I would not have a clue. Is there a better way to document it?

**Review by 4lejandrito — APPROVED** [review]

File: modules/apps/object/object-rest-impl/rest-openapi.yaml (hunk 1 above)
Comment (ccorreagg) [reply modules/apps/object/object-rest-impl/rest-openapi.yaml:34]: I guess by including the name of the nested field that must be used?. The description added is the one that is always included in the OpenAPI file. Maybe this is a special case and would need extra description. Or we could even think about updating dinamically the `nestedFields` description with the possible nested fields?

**Comment by ccorreagg** [comment]
Rebased with master

File: modules/apps/object/object-rest-impl/rest-openapi.yaml (hunk 1 above)
Comment (ccorreagg) [reply modules/apps/object/object-rest-impl/rest-openapi.yaml:34]: Well, I updated the PR with a new commit

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
executed buildREST

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by Hijky** [comment]
The failures are not related to your PR, it's okay to forward.

**Comment by ccorreagg** [comment]
Thanks @Hijky !

Sent here: https://github.com/brianchandotcom/liferay-portal/pull/142977

**Pull request closed without merging**